client/internal/engine.go

```diff
@@ -1474,7 +1482,20 @@ func (e *Engine) startNetworkMonitor() {
 	e.networkMonitor = networkmonitor.New()
 	go func() {
 		err := e.networkMonitor.Start(e.ctx, func() {
-			log.Infof("Network monitor detected network change, restarting engine")
+			log.Infof("Network monitor: detected network change")
+			e.lastNetworkChangeTimestamp.mu.Lock()
+			defer e.lastNetworkChangeTimestamp.mu.Unlock()
+			now := time.Now().Unix()
+			if e.lastNetworkChangeTimestamp.value+3 > now || e.cancel == nil {
```

Comment:
Maybe one second is enough?

Reply:
This doesn't add any additional delay. What it does is is stop restarting the engin if the last time it restarted the engine was less than 3 second. You could decrease it but it will not make bringing up the connection any faster